.github/workflows/pr.yml

```diff
@@ -3,16 +3,31 @@ on:
   pull_request:
     branches:
       - main
+env:
+  PR_TARGET_REF: "refs/remotes/origin/${{ github.event.pull_request.base.ref }}"
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout repo
         uses: actions/checkout@v4
+        with:
+          # Lerna needs *all* the git history to determine which packages have changed
+          fetch-depth: 0
+
+        # We need to fetch the target so that "lerna run <> --since origin/<>" works
+        # https://github.com/actions/checkout/issues/118#issuecomment-595438258
+      - name: Fetch target branch
+        run:  git fetch --no-tags --prune --depth=1 origin +refs/heads/${{ github.event.pull_request.base.ref }}:refs/remotes/origin/${{ github.event.pull_request.base.ref }}
 
-        # "prepare" hook on `bun install` runs `bun run build`
       - name: Setup environment
         uses: ./.github/actions/setup-env
+        with:
+          # Ignore lifecycle scripts to avoid building packages that haven't changed
+          ignore_scripts: true
+
+      - name: Build changed packages
+        run: bun run lerna run build --since ${{ env.PR_TARGET_REF }}
```

Comment:
Check if a package depends on another that has a build stage, and the dependency is not built because it is not changed - we need to make sure dependencies are built

Reply:
This is handled by the `nx.json` "^build" - packages build their dependencies before themselves